# origin: Debian (herbert)
# cset: n/a
# inclusion: not submitted
# description: allow setup-pci.c to use ide_scan_direction
# revision date: 2004-09-04

--- kernel-source-2.4.27-2.4.27/drivers/ide/setup-pci.c~	2004-09-04 14:47:31.000000000 -0700
+++ kernel-source-2.4.27-2.4.27/drivers/ide/setup-pci.c	2004-09-04 14:47:53.000000000 -0700
@@ -793,21 +793,20 @@
 
 /**
  *	ide_scan_pcibus		-	perform the initial IDE driver scan
- *	@scan_direction: set for reverse order scanning
  *
  *	Perform the initial bus rather than driver ordered scan of the
  *	PCI drivers. After this all IDE pci handling becomes standard
  *	module ordering not traditionally ordered.
  */
  	
-void __init ide_scan_pcibus (int scan_direction)
+void __init ide_scan_pcibus (void)
 {
 	struct pci_dev *dev;
 	struct pci_driver *d;
 	struct list_head *l, *n;
 
 	pre_init = 0;
-	if (!scan_direction) {
+	if (!ide_scan_direction) {
 		pci_for_each_dev(dev) {
 			ide_scan_pcidev(dev);
 		}
--- linux-2.4.27/include/linux/ide.h	2004-04-14 15:05:40.000000000 +0200
+++ linux-2.4.27+debian/include/linux/ide.h	2004-08-11 23:52:58.000000000 +0200
@@ -1254,6 +1254,7 @@
 
 #endif
 extern int noautodma;
+extern int ide_scan_direction;
 
 /*
  * We need blk.h, but we replace its end_request by our own version.
@@ -1614,7 +1615,7 @@
 extern int idescsi_attach(ide_drive_t *);
 extern int idescsi_init(void);
 
-extern void ide_scan_pcibus(int scan_direction) __init;
+extern void ide_scan_pcibus(void) __init;
 extern int ide_pci_register_driver(struct pci_driver *driver);
 extern void ide_pci_unregister_driver(struct pci_driver *driver);
 
--- kernel-source-2.4.27-2.4.27/drivers/ide/ide.c~	2004-09-04 14:46:54.000000000 -0700
+++ kernel-source-2.4.27-2.4.27/drivers/ide/ide.c	2004-09-04 14:47:22.000000000 -0700
@@ -172,7 +172,7 @@
 static int system_bus_speed;	/* holds what we think is VESA/PCI bus speed */
 static int initializing;	/* set while initializing built-in drivers */
 
-static int ide_scan_direction;	/* THIS was formerly 2.2.x pci=reverse */
+int ide_scan_direction;	/* THIS was formerly 2.2.x pci=reverse */
 
 #ifdef CONFIG_IDEDMA_AUTO
 int noautodma = 0;
@@ -2492,7 +2492,7 @@
 #ifdef CONFIG_BLK_DEV_IDEPCI
 	if (pci_present())
 	{
-		ide_scan_pcibus(ide_scan_direction);
+		ide_scan_pcibus();
 	}
 #endif /* CONFIG_BLK_DEV_IDEPCI */
 	ide_scan_drivers();
